src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,567 @@
+        storage_space_available = False
+        shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+            
+    except Exception as e:
+        logger.warning("An expection has occured while trying to fetch the events occured in azure-arc namespace from the cluster. Exception: {}".format(str(e)) + "\n")    
+    return storage_space_available
+
+
+def deployments_logger(appv1_api_instance, filepath_with_timestamp, storage_space_available):
+
+    try:
+        if storage_space_available :
+
+            # Creating new Deployment Logs folder in the given timestamp folder
+            deployments_path = os.path.join(filepath_with_timestamp, "Deployment_logs")
+            try:
+                os.mkdir(deployments_path)
+            except FileExistsError:
+                pass
+
+            # To retrieve all the the deployements that are present in the Cluster
+            deployments_list = appv1_api_instance.list_namespaced_deployment("azure-arc")
+
+            # Traversing through all the deployments present
+            for deployment in deployments_list.items:
+
+                # Fetching the deployment name
+                deployment_name = deployment.metadata.name
+
+                deployment_logs_path = os.path.join(deployments_path, deployment_name + ".txt")
+                # Creating a text file with the name of the deployment and adding deployment status in it
+                with open(deployment_logs_path, 'w+') as deployment_file:
+                    deployment_file.write(str(deployment.status))
+
+    except OSError as e:
+        storage_space_available = False
+        shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+            
+    except Exception as e:
+        logger.warning("An expection has occured while trying to fetch the azure arc deployment logs from the cluster. Exception: {}".format(str(e)) + "\n")
+    
+    return storage_space_available
+
+
+def check_agent_state(corev1_api_instance, filepath_with_timestamp, storage_space_available):
+
+    try:
+       
+        counter = 0
+        agent_state_path=os.path.join(filepath_with_timestamp, "Agent_State.txt")
+        with open(agent_state_path, 'w+') as agent_state:
+            # To retrieve all of the arc agent pods that are presesnt in the Cluster
+            arc_agents_pod_list = corev1_api_instance.list_namespaced_pod(namespace="azure-arc")
+
+            # Check if any arc agent other than kube aadp proxy is not in Running state
+            for each_agent_pod in arc_agents_pod_list.items:
+
+                if storage_space_available :
+
+
+                    # Storing the state of the arc agent in the user machine
+                    agent_state.write(each_agent_pod.metadata.name + " = " + each_agent_pod.status.phase +"\n")
+
+                # If the agent is Kube add proxy we will continue with the next agent
+                if(each_agent_pod.metadata.name.startswith("kube-aad-proxy")):
```

Comment:
why are we skipping this pod?

Reply:
In many cases the kube aad proxy maybe in container_creating or in pending state. But the reason for that is not less availability of resources. It may occur due to other reasons and we have added those in the check_security_policy and kap_cert check.